common/plugins/org.polarsys.capella.common.platform.sirius.customization/src/org/polarsys/capella/common/platform/sirius/customisation/SiriusCustomizationPlugin.java

```diff
@@ -0,0 +1,146 @@
+      SiriusEditPlugin.getPlugin().setUiCallback(new SiriusUiCallBack(SiriusEditPlugin.getPlugin().getUiCallback()));
+
+      customizeSiriusDefaultPreferences();
+    } catch (Exception e) {
+      getLog().log(new Status(IStatus.ERROR, PLUGIN_ID, IStatus.OK, e.getMessage(), e));
+    }
+  }
+
+  /**
+   * {@inheritDoc}
+   * 
+   * @see org.eclipse.ui.plugin.AbstractUIPlugin#stop(org.osgi.framework.BundleContext)
+   */
+  @Override
+  public void stop(BundleContext context) throws Exception {
+    plugin = null;
+    super.stop(context);
+  }
+
+  /**
+   * Returns the shared instance.
+   * 
+   * @return the shared instance
+   */
+  public static SiriusCustomizationPlugin getDefault() {
+    return plugin;
+  }
+
+  private void customizeSiriusDefaultPreferences() {
+    // ----------------
+    // Preference customization for plugin "org.eclipse.sirius"
+    IEclipsePreferences defaultScope = DefaultScope.INSTANCE.getNode(org.eclipse.sirius.viewpoint.SiriusPlugin.ID);
+    // Allow by default aird fragment with no representation creation
+    defaultScope.putBoolean(SiriusPreferencesKeys.PREF_EMPTY_AIRD_FRAGMENT_ON_CONTROL.name(), true);
+
+    // Re-apply command line customizations
+    applyCommandLineCustomizations(defaultScope);
+
+    // ----------------
+    // Preference customization for plugin "org.eclipse.sirius.common.ui"
+    defaultScope = DefaultScope.INSTANCE.getNode(SiriusTransPlugin.PLUGIN_ID);
+    // Disable Sirius Pre-commit listener behavior since Capella has the same one.
+    defaultScope.putBoolean(CommonPreferencesConstants.PREF_DEFENSIVE_EDIT_VALIDATION, false);
+
+    // Re-apply command line customizations
+    applyCommandLineCustomizations(defaultScope);
+
+    // ----------------
+    // Preference customization for plugin "org.eclipse.sirius.ui"
+    defaultScope = DefaultScope.INSTANCE.getNode(SiriusEditPlugin.ID);
+    // Required since Sirius 5.1 since default behavior is not welcome
+    defaultScope.putBoolean(SiriusUIPreferencesKeys.PREF_SAVE_WHEN_NO_EDITOR.name(), false);
+    defaultScope.putBoolean(SiriusUIPreferencesKeys.PREF_RELOAD_ON_LAST_EDITOR_CLOSE.name(), false);
+
+    // Set the default Sirius scale option (20%).
+    defaultScope.putInt(SiriusUIPreferencesKeys.PREF_SCALE_LEVEL_DIAGRAMS_ON_EXPORT.name(), 2);
+
+    // Don't use colors from odesign in diagram palettes
+    defaultScope.putBoolean(SiriusUIPreferencesKeys.PREF_DISPLAY_VSM_USER_FIXED_COLOR_IN_PALETTE.name(), false);
+
+    // Re-apply command line customizations
+    applyCommandLineCustomizations(defaultScope);
+  }
+
+  /**
+   * Method to re-apply the command line customizations done during {@link DefaultPreferences#load} after modification
+   * of default values done from an other plugin.
+   * 
+   * @param preferencesNode
+   *      the default preference node to customize.
+   */
+  private void applyCommandLineCustomizations(IEclipsePreferences preferencesNode) {
```

Comment:
@lfasani If I understand correctly, we need to do this dirty hack, because we modify at runtime the default preferences, so we need to ensure that these changes are reflected in the command line default preferences.

What if instead we remove all of this code from here, and we put all the preferences in `org.polarsys.capella.rcp\plugin_customization.ini`, **then we don't need to do this hack anymore right?** The command line preferences will be automatically syncronized.